sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -704,16 +814,91 @@ public HttpPipeline getHttpPipeline() {
             marker -> StorageImplUtils.applyOptionalTimeout(this.client.getMessages()
                 .dequeueWithResponseAsync(queueName, maxMessages, visibilityTimeoutInSeconds,
                     null, null, context), timeout)
-                .map(response -> new PagedResponseBase<>(response.getRequest(),
-                    response.getStatusCode(),
-                    response.getHeaders(),
-                    response.getValue(),
-                    null,
-                    response.getDeserializedHeaders()));
+                .flatMap(this::transformMessagesDequeueResponse);
 
         return new PagedFlux<>(() -> retriever.apply(null), retriever);
     }
 
+    private Mono<PagedResponseBase<MessagesDequeueHeaders, QueueMessageItem>> transformMessagesDequeueResponse(
+        MessagesDequeueResponse response) {
+        List<QueueMessageItemInternal> queueMessageInternalItems = response.getValue();
+        if (queueMessageInternalItems == null) {
+            queueMessageInternalItems = Collections.emptyList();
+        }
+        return Flux.fromIterable(queueMessageInternalItems)
+            .flatMap(queueMessageItemInternal ->
+                transformQueueMessageItemInternal(queueMessageItemInternal, messageEncoding)
+                .onErrorResume(IllegalArgumentException.class, e -> {
+                    if (processMessageDecodingErrorAsyncHandler != null) {
+                        return transformQueueMessageItemInternal(
+                            queueMessageItemInternal, QueueMessageEncoding.NONE)
+                            .flatMap(messageItem -> processMessageDecodingErrorAsyncHandler.apply(
+                                new QueueMessageDecodingError(
+                                    this, new QueueClient(this),
+                                    messageItem, null, e)))
+                            .then(Mono.empty());
```

Comment:
nit: `then(Mono.empty())` isn't required as `processMessageDecodingErrorAsyncHandler` returns `Mono<Void>` which is the same.

Reply:
Expected return type from this closure is `Mono<PeekedMessageItem>` . If I remove this line it fails to compile.
When badly encoded message is handled by handler we want it to be **not** included in receive/peek api response, hence emiting empty from here (this whole chain processes Flux<PeekedMessageItemInternal> to Flux<PeekedMessageItem>